### ISSUE TITLE
Issues with TextToSpeechV1

### ISSUE THREAD

**Issue by s2t2** [watson-developer-cloud/python-sdk#639]
#### Expected behavior

Ultimately I would like to be able to use `TextToSpeechV1.get_word()` to detect phonemes from some given text. 

After [some initial confusion](https://stackoverflow.com/questions/54599355/ibm-phoneme-detection-in-python) about how to pass a valid `customization_id` parameter to that method, I think I need to use `TextToSpeechV1.create_voice_model()` to create a custom voice model, and use that model's identifier as the `customization_id` value. If this is the correct strategy, I could use help debugging the implementation below. Otherwise if this is not the right strategy, I would appreciate some pointers in the right direction. Thanks!

See also:

  + [Method definition](https://github.com/watson-developer-cloud/python-sdk/blob/b23ab79a6bf3bdc8e3fb66267eee2b2d2f8b5e66/watson_developer_cloud/text_to_speech_v1.py#L362-L386)
  + [API documentation](https://cloud.ibm.com/apidocs/text-to-speech?code=python#create-a-custom-model)

#### Actual behavior

The call to create a new voice model results in an API error: "ERROR 400: Required parameter 'base_model_name' is missing". 

#### Steps to reproduce the problem

See code below...

#### Code snippet (Note: Do not paste your credentials)

```py
from watson_developer_cloud import TextToSpeechV1, WatsonApiException

API_KEY = "<redacted>"
URL = "https://gateway-wdc.watsonplatform.net/text-to-speech/api"

client = TextToSpeechV1(iam_apikey=API_KEY, url=URL)

LANG = "en-US" # Allowable values: [de-DE, en-US, en-GB, es-ES, es-LA, es-US, fr-FR, it-IT ,ja-JP, pt-BR]

try:
    # INTERMEDIATE STEP

    voice_model_response = client.create_voice_model(
        name="My Custom Model",
        language=LANG,
        description="to get a valid 'customization_id' value..."
    ).get_result() #> ERROR 400: Required parameter 'base_model_name' is missing

    customization_id = voice_model_response["customization_id"]

    # END GOAL

    response = client.get_word(customization_id=customization_id, word="HELLO WORLD")

except WatsonApiException as ex:
    print(f"ERROR {str(ex.code)}: {ex.message}")
```

#### python sdk version

sorry, not sure how to get this. i just recently installed `watson_developer_cloud` via pip, so probably some recent stable release.

#### python version

Python 3.7.0

**Comment by ehdsouza** [comment]
@s2t2 your code for creating voice model seems correct, I just tested your code for creating voice model and it gave a response:
```
voice_model_response = service.create_voice_model(
    name="My Custom Model",
    language="en-US",
    description="to get a valid 'customization_id' value..."
).get_result()
print(json.dumps(voice_model_response, indent=2))
```
response:
```json
{
  "customization_id": "9f843339-aac7-4dca-baad-641d16bb254a"
}
```

Looks like another line in your code is throwing the error. Can you confirm?

**Comment by s2t2** [comment]
Hi @ehdsouza Thanks for your response. Unfortunately I don't see the same result. Here is the code that causes the error for me:

```py
request = client.create_voice_model(name="My Custom Model", language="en-US", description="to get a valid 'customization_id' value...").get_result()

#> *** watson_developer_cloud.watson_service.WatsonApiException: 
#> Error: Required parameter 'base_model_name' is missing, Code: 400
#> Information: {'warnings': ['Unexpected input parameter(s) [language] detected in the passed JSON'], 'code_description': 'Bad Request'} , X-dp-watson-tran-id: redacted , X-global-transaction-id: redacted
```

Note, this happens also without the trailing `get_result()`. Please advise.

**Comment by ehdsouza** [comment]
@s2t2 could you try with a curl to see if response is the same?

**Comment by s2t2** [comment]
Unfortunately I'm seeing similar unsuccessful results (using two different possible base URL options ("stream.watsonplatform.net" vs "gateway-wdc.watsonplatform.net"):

Ideal Goal of parsing text into speech parts / phonemes:

```
curl -X GET -u "apikey:REDACTED" "https://stream.watsonplatform.net/text-to-speech/api/v1/pronunciation?text=IEEE&format=ibm&voice=en-US_AllisonVoice"
#> {"code":401, "error": "Unauthorized"}

curl -X GET -u "apikey:REDACTED" "https://gateway-wdc.watsonplatform.net/speech-to-text/api/v1/pronunciation?text=IEEE&format=ibm&voice=en-US_AllisonVoice"
#> {"error": "No such child resource.",  "code": 404, "code_description": "No Such Resource"}
```

Potential Intermediate Step of creating a custom voice model first:

```sh
curl -X POST -u "apikey:REDACTED" --header "Content-Type: application/json" --data "{\"name\":\"My CURL Model\", \"language\":\"en-US\", \"description\":\"First custom voice model\"}" "https://stream.watsonplatform.net/text-to-speech/api/v1/customizations"
#> {"code":401, "error": "Unauthorized"}

curl -X POST -u "apikey:REDACTED" --header "Content-Type: application/json" --data "{\"name\":\"My CURL Model\", \"language\":\"en-US\", \"description\":\"First custom voice model\"}" "https://gateway-wdc.watsonplatform.net/speech-to-text/api/v1/customizations"
#> {
#>    "code": 400,
#>    "warnings": ["Unexpected input parameter(s) [language] detected in the passed JSON"],
#>    "code_description": "Bad Request",
#>    "error": "Required parameter 'base_model_name' is missing"
#> }
```

**Comment by germanattanasio** [comment]
@daniel-bolanos can you look into this ☝️

**Comment by daniel-bolanos** [comment]
hi @s2t2, I just took a quick look at your curl commands:

The first one has bad credentials, so that needs to be fixed first.

```
curl -X GET -u "apikey:REDACTED" "https://stream.watsonplatform.net/text-to-speech/api/v1/pronunciation?text=IEEE&format=ibm&voice=en-US_AllisonVoice"
#> {"code":401, "error": "Unauthorized"}
```

The second one is passing the `speech-to-text` path instead of the tts one.

```
curl -X GET -u "apikey:REDACTED" "https://gateway-wdc.watsonplatform.net/speech-to-text/api/v1/pronunciation?text=IEEE&format=ibm&voice=en-US_AllisonVoice"
#> {"error": "No such child resource.",  "code": 404, "code_description": "No Such Resource"} 
```

**Comment by s2t2** [comment]
Thanks @daniel-bolanos for your reply. After updating the URL, I was able to see both responses as "Unauthorized".  I thought I had been using the right API key, but I created and upgraded another TTS service from scratch and that service's API Key seems to be working now. 🙌 

So now I'm seeing 400 errors relating to `"[WORDS] not found in custom model [CUSTOMIZATION_ID]"`. But that's probably more a matter of me needing to further consult the API docs, so I think we can close this issue. 

Thanks everyone for your help!
